spec/app/domain/authentication/authn-azure/application_identity_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Authentication::AuthnAzure::ApplicationIdentity do
+  include_context "azure setup"
+
+  let(:user_assigned_identity_service_id_scoped_annotation) { double("UserAssignedIdentityAnnotation") }
+  let(:system_assigned_identity_service_id_scoped_annotation) { double("UserAssignedIdentityServiceIdScopeAnnotation") }
+
+  before(:each) do
+    define_host_annotation(host_annotation_type: user_assigned_identity_service_id_scoped_annotation, host_annotation_key: "#{granular_annotation_type}/user-assigned-identity", host_annotation_value: "some-user-assigned-service-id-scoped-value")
+    define_host_annotation(host_annotation_type: system_assigned_identity_service_id_scoped_annotation, host_annotation_key: "#{granular_annotation_type}/system-assigned-identity", host_annotation_value: "some-system-assigned-service-id-scoped-value")
+
+    define_host_annotation(host_annotation_type: non_azure_annotation, host_annotation_key: "#{global_annotation_type}/non-azure-annotation", host_annotation_value: "some-non-azure-value")
+  end
+
+  context "An application identity in annotations" do
+    subject(:application_identity) {
```

Comment:
why do you write `subject(:application_identity)` if you are referring to it later as `subject`? choose one approach and stick with it

Reply:
Choose one between:
```
subject(:application_identity) {
      Authentication::AuthnAzure::ApplicationIdentity.new(
        role_annotations: role_annotations,
        service_id:       test_service_id
      )
    }
    context("with a global scoped constraint") do
      let(:role_annotations) { [subscription_id_annotation, resource_group_annotation, user_assigned_identity_annotation, system_assigned_identity_annotation] }

      it "Returns Hash of the constraint and its value" do
        expect(application_identity.constraints).to eq(
                                         { subscription_id:          "some-subscription-id-value",
                                           resource_group:           "some-resource-group-value",
                                           user_assigned_identity:   "some-user-assigned-identity-value",
                                           system_assigned_identity: "some-system-assigned-identity-value"
                                         })
      end
    end
```
```
subject  do
      Authentication::AuthnAzure::ApplicationIdentity.new(
        role_annotations: role_annotations,
        service_id:       test_service_id
      )
    end
    context("with a global scoped constraint") do
      let(:role_annotations) { [subscription_id_annotation, resource_group_annotation, user_assigned_identity_annotation, system_assigned_identity_annotation] }

      it "Returns Hash of the constraint and its value" do
        expect(subject.constraints).to eq(
                                         { subscription_id:          "some-subscription-id-value",
                                           resource_group:           "some-resource-group-value",
                                           user_assigned_identity:   "some-user-assigned-identity-value",
                                           system_assigned_identity: "some-system-assigned-identity-value"
                                         })
      end
    end
```

Note that if you give the subject a name then you can use that name instead of subject in `it` clauses